tests/components/yessssms/test_notify.py

```diff
@@ -151,210 +149,219 @@ async def test_connection_error_on_init(hass, caplog, valid_settings, connection
+    )
+
+    message = "Testing YesssSMS platform :)"
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 1
+
+
+async def test_empty_message_error(yessssms, caplog):
+    """Test for an empty SMS message error."""
+    message = ""
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+
+    for record in caplog.records:
+        if (
+            record.levelname == "ERROR"
+            and record.name == "homeassistant.components.yessssms.notify"
+        ):
+            assert "Cannot send empty SMS message" in record.message
+
+
+async def test_error_account_suspended(yessssms, requests_mock, caplog):
+    """Test login that fails after multiple attempts."""
+    requests_mock.post(
+        # pylint: disable=protected-access
+        yessssms.yesss._login_url,
+        status_code=200,
+        text="BlaBlaBla<strong>Login nicht erfolgreichBlaBla",
+    )
+
+    message = "Testing YesssSMS platform :)"
+
+    yessssms.send_message(message)
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 1
+
+    requests_mock.post(
         # pylint: disable=protected-access
-        self.yessssms.yesss._suspended = True
-
-        with self.assertLogs(
-            "homeassistant.components.yessssms.notify", level="ERROR"
-        ) as context:
-            self.yessssms.send_message(message)
-        self.assertIn("Account is suspended, cannot send SMS.", context.output[0])
-
-    @requests_mock.Mocker()
-    def test_send_message(self, mock):
-        """Test send message."""
-        message = "Testing YesssSMS platform :)"
-        mock.register_uri(
-            "POST",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._login_url,
-            status_code=302,
-            # pylint: disable=protected-access
-            headers={"location": self.yessssms.yesss._kontomanager},
-        )
+        yessssms.yesss._login_url,
+        status_code=200,
+        text="Wegen 3 ungültigen Login-Versuchen ist Ihr Account für "
+        "eine Stunde gesperrt.",
+    )
+
+    message = "Testing YesssSMS platform :)"
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 2
+
+
+async def test_error_account_suspended_2(yessssms, caplog):
+    """Test login that fails after multiple attempts."""
+    message = "Testing YesssSMS platform :)"
+    # pylint: disable=protected-access
+    yessssms.yesss._suspended = True
+
+    with caplog.at_level(logging.ERROR):
+        yessssms.send_message(message)
+    for record in caplog.records:
+        if (
+            record.levelname == "ERROR"
+            and record.name == "homeassistant.components.yessssms.notify"
+        ):
+            assert "Account is suspended, cannot send SMS." in record.message
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)